src/contradictory_claims/data/preprocess_cord.py

```diff
@@ -27,7 +34,147 @@ def filter_metadata_for_covid19(metadata_path: str, virus_lex_path: str, pub_dat
+        list_of_filenames = zipobj.namelist()
+        # print('Number of files to iterate over:',len(list_of_filenames))
+        k = 0
+        # iter_num = 0
+        for filename in list_of_filenames:
+            # iter_num = iter_num + 1
+            # Check filename ends with json and file exists in filtered list of cord papers
+            if (filename in pdf_filenames) or (filename in pmc_filenames):
+                zipobj.extract(filename, json_temp_path)
+                json_op = []
+                with open(json_temp_path + filename, 'r', encoding='utf8') as f:
+                    temp = []
+                    temp.append("".join([line.replace('\n', '').replace('\r', '').replace('\t', '') for line in f]))
+                    for jsonobj in temp:
+                        json_dict = json.loads(jsonobj, encoding='utf8')
+                        json_op.append(json_dict)
+                        for i in range(len(json_op)):
+                            temp_json = json_normalize(json_op[i])
+                            check_file_name = ((filename == covid19_metadata.pdf_json_files)
+                                               | (filename == covid19_metadata.pmc_json_files))  # noqa: W503
+                            cord_uid = list(covid19_metadata.loc[check_file_name, 'cord_uid'])[0]
+                            try:
+                                text = temp_json['abstract'][0][0]['text']
+                                section = temp_json['abstract'][0][0]['section']
+                                for key, v in replace_dict.items():
+                                    text = text.replace(key, v)
+                                    section = section.replace(key, v)
+                                covid19_dict[k] = {'cord_uid': cord_uid,
+                                                   'sentence': text,
+                                                   'section': section}
+                                k = k + 1
+                            except KeyError:
+                                pass
+
+                            for temp_dict in temp_json['body_text'][0]:
+                                text = temp_dict['text']
+                                section = temp_dict['section']
+                                for key, v in replace_dict.items():
+                                    text = text.replace(key, v)
+                                    section = section.replace(key, v)
+                                covid19_dict[k] = {'cord_uid': cord_uid,
+                                                   'sentence': text,
+                                                   'section': section}
+                                k = k + 1
+
+#        if iter_num%100==0:
+#            print('Number of files read:', iter_num)
+
+    return pd.DataFrame.from_dict(covid19_dict, orient='index')
+
+
+def construct_regex_match_pattern(search_terms: List[str], search_type: bool = 0):
```

Comment:
oh woah you can make 1 and 0 bools? What about the variable `exact_search: bool = True` instead? That seems clearer and maybe more standard

Reply:
Yeah you can. On second thought, I'll use string instead of bool.